File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
selected = []

def solution(numbers, target):    
    def dfs(currentValue, depth):  
        if depth == len(numbers):
            selected.append(currentValue)
            return
        
        for oper in ['+', '-']:
            if oper == '+':
                dfs(currentValue + numbers[depth] * 1, depth + 1)
            else:
                dfs(currentValue - numbers[depth] * 1, depth + 1)
                
    dfs(0, 0)
    answer = selected.count(target)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
	const N = numbers.length;

	let answer = 0;
	let selected = [];
	function DFS(n) {
		if (n === N) {
			let expression = '';
			for (let i = 0; i < N; i++) {
				expression += selected[i] + numbers[i];
			}
			if (eval(expression) === target) answer++;
		} else {
			for (let oper of ['+', '-']) {
				selected.push(oper);
				DFS(n + 1);
				selected.pop();
			}
		}
	}

	DFS(0);

	return answer;
}